server/amp/helpers/infinite-scroll.js

```diff
@@ -36,39 +44,51 @@ class InfiniteScrollAmp {
 
   getImagePath(item) {
     const cdnImage = this.publisherConfig["cdn-image"];
-    const s3Key = item.story["hero-image-s3-key"];
+    const s3Key = item["hero-image-s3-key"];
     const hostWithProtocol = /^https:\/\//.test(cdnImage) ? cdnImage : `https://${cdnImage}`;
     return `${hostWithProtocol}/${s3Key}?format=webp&w=250`;
   }
 
-  async getResponse({ itemsTaken }) {
+  async getInfiniteScrollList({ storyId, type, offset = 0, limit = null }) {
+    let filteredItems = [];
+    const params = { offset, limit };
```

Comment:
Can take limit, offset only if they have truthy value like this:
```js
const params = { 
  ...(limit && {limit}),
  ...(offset && {offset})
}
```
https://stackoverflow.com/a/40560953